packages/frontend-core/src/components/grid/cells/BBReferenceCell.svelte

```diff
@@ -0,0 +1,35 @@
+<script>
+  import { getContext } from "svelte"
+  import RelationshipCell from "./RelationshipCell.svelte"
+  import { FieldSubtype } from "@budibase/types"
+
+  const { API } = getContext("grid")
+  const { subtype } = $$props.schema
+
+  const schema = {
+    ...$$props.schema,
+    tableId: "external",
```

Comment:
If we just change this to be the users table ID (or dynamically set based on subtype), then the relationship cell should correctly use this and fetch the definition of the users table and pull out the correct primary display column.

Would that not let us only require one new custom prop (`searchFunction`) rather than 2? I also think that currently the relationship cell will override the value of primary display when it uses `getPrimaryDisplayForTableId` to try and get the correct table ID for the schema, which is currently `external`, and I would expect that to throw an error.

If that works then that would be great since it would be nice to not have to manually override the primary display field.

Reply:
The issue in here is that the relationship currently is looking for app tables, and this will be a global table. Another API is required, for this I did override the API. And the table id is actually not used in here, but before this I was getting some rendering errors when we were checking `if !tableId`

Reply:
Maybe we could modify the external component to have "global vs app data", and we use the global API or the table API. It might cover most of the cases in the future

Reply:
I guess since we're overriding the search function, it's alright to override primary display too. If we were going to add awareness of this global users table then we would probably allow it to understand how to search that table too, so thats fine.

In that case, we can leave it like it is here 👍

Reply:
If we need to extend this in the future we can pass a parameter to differentiate server vs global tables, but I wanted to keep the changes to the RelationshipCell as minimal as possible